src/shared/clients/defaultEcsClient.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ECS } from 'aws-sdk'
+import { ext } from '../extensionGlobals'
+import { EcsClient } from './ecsClient'
+
+export class DefaultEcsClient implements EcsClient {
```

Comment:
It would be nice to get testing coverage on this class.
Changing `createSdkClient` to protected allows us to make a `TestEcsClient` class, which serves up a mock client, enabling us to verify the function behaviors.

Reply:
We don't have any test coverage on any of the other clients, which isn't an excuse, but is a precedent. I'll see what I can do, but we may want to cut a separate issue for it.

Reply:
Done.

Reply:
Requesting test coverage for other clients here: https://github.com/aws/aws-toolkit-vscode/issues/718